### PR TITLE
Homework 

### PR DESCRIPTION
**Pull request by D0LeD**

### CODE REVIEW

File: exercises/any_odds_and_evens/main_test.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	}{
+		{[]int{}, false},
+		{[]int{0}, false},
+		{[]int{2}, false},
+		{[]int{3}, true},
+		{[]int{4, 2}, false},
+		{[]int{2, 3}, true},
+		{[]int{3, 1, 5}, true},
+		{[]int{2, 4, 1}, true},
+		{[]int{-1}, true},
```
Comment (yarcat) [inline exercises/any_odds_and_evens/main_test.go:18]: nit: consider some odd/even negatives as well

File: exercises/any_odds_and_evens/main_test.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+		{[]int{2}, true},
+		{[]int{2, 8}, true},
+		{[]int{3, 5}, false},
+		{[]int{2, 7}, true},
+		{[]int{3, 1, 5}, false},
+		{[]int{2, 4, 6}, true},
+		{[]int{-1}, false},
+	} {
+		got := AnyIsEven(tc.s)
+		if got != tc.want {
```
Comment (yarcat) [inline exercises/any_odds_and_evens/main_test.go:43]: nit: consider writing these lines as one:
```go
if got := AnyIsEven(ts.s); got != tc.want {
  ...
}
```

File: exercises/filter_odds_and_evens/main_test.go [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+	for _, tc := range []struct {
+		s    []int
+		want []int
+	}{
+		{[]int{}, []int{}},
+		{[]int{0}, []int{0}},
+		{[]int{0, 1}, []int{0}},
+		{[]int{-1, -2}, []int{-2}},
+		{[]int{3, 7, 4, 9}, []int{4}},
+		{[]int{4, 6, 8, 1, 3, 91}, []int{4, 6, 8}},
```
Comment (yarcat) [inline exercises/filter_odds_and_evens/main_test.go:15]: nit: consider some negatives here as well

File: exercises/only_odds_and_evens/main_test.go [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+		want bool
+	}{
+		{[]int{}, true},
+		{[]int{0}, false},
+		{[]int{1}, true},
+		{[]int{2}, false},
+		{[]int{2, 4}, false},
+		{[]int{1, 6}, false},
+		{[]int{1, 3, 5, 9, 6}, false},
+		{[]int{-1}, true},
```
Comment (yarcat) [inline exercises/only_odds_and_evens/main_test.go:17]: nit: Consider adding more cases with negatives

File: exercises/rect/main_test.go [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+	"reflect"
+	"testing"
+)
+
+func TestRect(t *testing.T) {
+	for _, tc := range []struct {
+		x, y int
+		want [][]int
+	}{
+		{0, 0, [][]int{}},
```
Comment (yarcat) [inline exercises/rect/main_test.go:13]: nit: consider adding `{0, X}` and `{X, 0}` (where `X > 0`). Also negatives

File: exercises/spiral/main_test.go [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+	"reflect"
+	"testing"
+)
+
+func TestSpiral(t *testing.T) {
+	for _, tc := range []struct {
+		x, y int
+		want [][]int
+	}{
+		{0, 0, [][]int{}},
```
Comment (yarcat) [inline exercises/spiral/main_test.go:13]: nit: consider adding tests for (0, x), (x, 0) and negative inputs

File: exercises/spiral/main_test.go [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+		{1, 1, [][]int{
+			{1},
+		}},
+		{3, 5, [][]int{
+			{1, 2, 3},
+			{12, 13, 4},
+			{11, 14, 5},
+			{10, 15, 6},
+			{9, 8, 7},
+		}},
```
Comment (zayac) [inline exercises/spiral/main_test.go:23]: Based on the input, there should be `3` rows and `5` columns, not the opposite. Could you please change it?

File: exercises/spiral/main.go [hunk 8]
```diff
@@ -0,0 +1,35 @@
+package main
+
+import "fmt"
+
+func Spiral(x, y int) [][]int {
```
Comment (zayac) [inline exercises/spiral/main.go:5]: This is wrong implementation. The tests are failing. You can check this by running `go test -v ./exercises/spiral`.
